dspace-server-webapp/src/test/java/org/dspace/app/rest/csv/CSVMetadataImportReferenceTest.java

```diff
@@ -0,0 +1,408 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.csv;
+
+import static junit.framework.TestCase.assertEquals;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.UUID;
+
+import org.dspace.app.rest.builder.CollectionBuilder;
+import org.dspace.app.rest.builder.CommunityBuilder;
+import org.dspace.app.rest.builder.ItemBuilder;
+import org.dspace.app.rest.test.AbstractEntityIntegrationTest;
+import org.dspace.content.Collection;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataField;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.Relationship;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.ItemService;
+import org.dspace.content.service.MetadataFieldService;
+import org.dspace.content.service.MetadataValueService;
+import org.dspace.content.service.RelationshipService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Created by: Andrew Wood
+ * Date: 26 Jul 2019
+ */
+public class CSVMetadataImportReferenceTest extends AbstractEntityIntegrationTest {
+
+    //Common collection to utilize for test
+    private Collection col1;
+
+    @Autowired
+    private RelationshipService relationshipService;
+
+    @Autowired
+    private ItemService itemService;
+
+    /**
+     * Setup testing enviorment
+     */
+    @Before
+    public void setup() {
+        context.turnOffAuthorisationSystem();
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                .withName("Parent Community")
+                .build();
+        col1 = CollectionBuilder.createCollection(context, parentCommunity).withName("Collection 1").build();
+    }
+
+    /**
+     * Close testing enviorment and restore auth system
+     */
+    @After
+    public void close() {
+        context.restoreAuthSystemState();
```

Comment:
@AndrewZWood : Just a clarification question. Is there a reason why we need authorization turned *off* while running *all tests*?  To me, it looks like our tests are all being kicked off via the commandline (`runDSpaceScript()` method), which means that we should be creating a new Context for each test anyways.  This implies to me that we might be able to do the `restoreAuthSystemState();` *at the very end of the `setup()` method above*.  In other words, it seems like we only need authorization turned off to create the Community/Collection in the `setup()`.  Am I missing something though?

Reply:
This is a good point, I'll be refactoring `CSVMetadataImportReferenceTest.java` as a part of [DS-4226](https://jira.duraspace.org/browse/DS-4226) and I've noted to verify context authorization requirements at a per test basis.